js/src/Ice/Communicator.js

```diff
@@ -2,25 +2,20 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
-const Ice = require("../Ice/ModuleRegistry").Ice;
-
-require("../Ice/AsyncResultBase");
-require("../Ice/Debug");
-require("../Ice/Instance");
-require("../Ice/LocalException");
-require("../Ice/UUID");
-
-const Instance = Ice.Instance;
-const Debug = Ice.Debug;
+import { CommunicatorDestroyedException } from "./LocalException.js";
+import { generateUUID } from "./UUID.js";
+import { identityToString } from "./IdentityUtil.js";
+import { Promise } from "./Promise.js";
+import { Debug } from "./Debug.js";
```

Comment:
Importing Instance and AsyncResulBase result in cicular dependencies.  Moved the dependencies to `CommunicatorExtensions.js` which defines `createAsyncResultBase` and `createInstance` methods.